### PR TITLE
pull request for user-service project

### PR DESCRIPTION
**Pull request by mStack5522**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -28,4 +28,19 @@ eurekaclinical.userservice.email.verify.subject=Eureka! Clinical Analytics Regis
 eurekaclinical.userservice.email.activation.subject=Eureka! Clinical Analytics Account Activation
 eurekaclinical.userservice.email.password.subject=Password changed on Eureka! Clinical Analytics
 eurekaclinical.userservice.email.reset.subject=Password reset on Eureka! Clinical Analytics
-eurekaclinical.userservice.registration.timeout=72
\ No newline at end of file
+eurekaclinical.userservice.registration.timeout=72
+
+eurekaclinical.userservice.localregistrationenabled=true
+eurekaclinical.userservice.githuboauthcallbackurl=https://localhost:4200/oauthcallback/github
+eurekaclinical.userservice.githuboauthkey=6848fbb612df8f5c5a15
```
Comment (arpost) [inline src/main/resources/application.properties]: I doubt you really want these keys and secrets in there. This is a public repo.

File: nb-configuration.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (arpost) [inline nb-configuration.xml]: The nb-configuration.xml file should be in your global gitignore so it does not go into version control.

File: pom.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <groupId>org.eurekaclinical</groupId>    
     <artifactId>eurekaclinical-user-service</artifactId>
     <packaging>war</packaging>
-    <version>1.0-Alpha-32-SNAPSHOT</version>
+    <version>1.0-Alpha-33</version>
```
Comment (arpost) [inline pom.xml]: There's something wrong with this version number. Only snapshots should be going into upstream-master.

File: src/main/java/org/eurekaclinical/user/service/resource/GitHubRegistrationOAuthUserResource.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package org.eurekaclinical.user.service.resource;
+
+import com.google.inject.Inject;
```
Comment (arpost) [inline src/main/java/org/eurekaclinical/user/service/resource/GitHubRegistrationOAuthUserResource.java]: Please replace with javax.inject.Inject if possible.

File: src/main/java/org/eurekaclinical/user/service/resource/GlobusRegistrationOAuthUserResource.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package org.eurekaclinical.user.service.resource;
+
+import com.google.inject.Inject;
```
Comment (arpost) [inline src/main/java/org/eurekaclinical/user/service/resource/GlobusRegistrationOAuthUserResource.java]: Please replace with javax.inject.Inject if possible.

File: src/main/java/org/eurekaclinical/user/service/resource/GoogleRegistrationOAuthUserResource.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package org.eurekaclinical.user.service.resource;
+
+import com.google.inject.Inject;
```
Comment (arpost) [inline src/main/java/org/eurekaclinical/user/service/resource/GoogleRegistrationOAuthUserResource.java]: Please replace with javax.inject.Inject if possible.

File: src/main/java/org/eurekaclinical/user/service/util/AbstractOAuthRegistrationUserResource.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        this.provider = oauthProvider;
+    }
+    
+    
+    
+    public E getEurekaProfile(HttpServletRequest req) throws HttpStatusException {
+        assert req != null : "request cannot be null";
+        OAuthCredential credential
+                = this.provider.getCredential(
+                        new HttpUserSession(req.getSession()),
```
Comment (arpost) [inline src/main/java/org/eurekaclinical/user/service/util/AbstractOAuthRegistrationUserResource.java:53]: Please confirm that we're not storing any user information in the session as a result of the OAuth workflow.

**Comment by arpost** [comment]
Thanks! I have a few minor change requests, and a security-related question, in the comments.

**Comment by mStack5522** [comment]
Np, I will look into it and update accordingly, thanks.

**Pull request merged**